### PR TITLE
WIP: Reviving dhadron case with Lc PbPb 080

### PR DESCRIPTION
**Pull request by lvermunt**
@benedikt-voelkel 

A working database for Lc PbPb 0-80% (for 2015). I commented all the non relevant parts in the database which are not needed for this analysis. So it complains when running and we can change the line into a `database.get("blabla", None)`. So far I only did the converting stage. As it is GRID applied, there are now some modifications needed.

Let me know if you have comments

### CODE REVIEW

**Comment by ginnocen** [comment]
hi @LuukVermunt @benedikt-voelkel please always ask for review whenever you touch something on top of the "inclusive" classes to make sure we dont mess up with the rest.

**Comment by lvermunt** [comment]
Hi @ginnocen, sure, but note this is not yet going into `master`. We are working in a dev branch till everything is working properly

**Comment by benedikt-voelkel** [comment]
Hej @ginnocen,

adding to @lvermunt 's comment:
1. The inclusive analyzer and processer in `master` don't work at the moment (not compatible with DBs), so from that point of view we cannot not break anything if we work in `analysis-dev`
2. nothing else in the package relies on the inclusive processer and analyzer, right?

So we should be pretty safe here, right? :)

But I will put another PR on top later which should be reviewed and discussed also on Thursday. And of course I will come back to you about that

File: machine_learning_hep/processing/multiprocesser.py [hunk 1]
```diff
@@ -46,7 +46,8 @@ def __init__(self, case, proc_class, datap, typean, run_param, mcordata):
         self.d_pklml_mergedallp = datap["multi"][self.mcordata]["pkl_skimmed_merge_for_ml_all"]
         self.d_pklevt_mergedallp = datap["multi"][self.mcordata]["pkl_evtcounter_all"]
 
-        self.dlper_mcreweights = datap["multi"][self.mcordata]["mcreweights"]
+        self.dlper_mcreweights = datap["multi"][self.mcordata].get("mcreweights", \
+                                               [None for _ in range(len(self.p_period))])
```
Comment (benedikt-voelkel) [inline machine_learning_hep/processing/multiprocesser.py:50]: Could do like
```
a = b.get("some", [None] * len(other))
```

File: machine_learning_hep/processing/processer.py [hunk 2]
```diff
@@ -49,7 +49,7 @@ def __init__(self, case, datap, run_param, mcordata, p_maxfiles,
         #self.logger = get_logger()
         self.processer_helper = ProcesserHelper(datap, case, typean, p_period)
         self.nprongs = datap["nprongs"]
-        self.prongformultsub = datap["prongformultsub"]
+        self.prongformultsub = datap.get("prongformultsub", [0 for _ in range(self.nprongs)])
```
Comment (benedikt-voelkel) [inline machine_learning_hep/processing/processer.py:52]: See above

**Comment by benedikt-voelkel** [comment]
Hej @lvermunt 

So when you do the application on the grid you still need to run all processer (except for application) and analyzer steps right?

**Comment by lvermunt** [comment]
Hi @benedikt-voelkel. Yes, so converging and skimming still needs to be done. The skimmed files are the input to the analyzer, where the ML probability column has a different name (`ml_prob`). I will try some things now and push into this PR.

Thanks for the suggestions! Forgot to write it, but with "comments" I also mentioned these code-in-a-nicer-way points :)

File: machine_learning_hep/analysis/analyzer_jet.py [hunk 3]
```diff
@@ -24,7 +24,10 @@
 # pylint: disable=import-error, no-name-in-module
 import uproot
 from ROOT import TFile, TH1F, TH2F, TCanvas, TLatex, TGraphAsymmErrors, TLine, TGaxis
-from ROOT import AliHFInvMassFitter, AliVertexingHFUtils
+try:
+    from ROOT import AliHFInvMassFitter, AliVertexingHFUtils
+except ImportError:
+    print("\n\033[1mFutureWarning: not in AliPhysics environment. Analyzer will crash!\033[0m\n")
```
Comment (benedikt-voelkel) [inline machine_learning_hep/analysis/analyzer_jet.py:30]: This should maybe also  not be a "FutureWarning" cause that would be something which refers to something coming/changing in the future, e.g. introducing changes to the API. 
How about just warn and put a `sys.exit(1)`?

**Comment by benedikt-voelkel** [comment]
Hello @lvermunt 
I was thinking about the different column name for the ML prediction from the grid and I think it makes sense to be able to specify a name for that explicitly. So if nothing is given, it uses the package default name, otherwise it takes the one in the database. Or do you think we shouldn't add another parameter in the DB's analysis section?

File: machine_learning_hep/analysis/analyzer_jet.py (hunk 3 above)
Comment (lvermunt) [reply machine_learning_hep/analysis/analyzer_jet.py:30]: As discussed on Mattermost, `sys.exit(1)` is not the way to go here. Idea now is to introduce a decorator (`@requiresaliphysics`). Issue to be opened

**Comment by ginnocen** [comment]
Hi @lvermunt. would be good to merge this PR, so that also Clara and Jaime can make use of it. what is the status? thanks GM

**Comment by ginnocen** [comment]
Hi @LuukVermunt. Can we try to complete this PR even in some preliminary form so that we don't get out of sync with the other developments?

**Comment by lvermunt** [comment]
Hi @ginnocen,

This PR is far from ready to be merged. I will close it, since I haven't had time to work on it in the last weeks. Note that it is also on top of ginnocen:analysis-dev, which @benedikt-voelkel and I started to use for some big restructuring of the processer and analyser classes this summer. Those developments are not yet ready, making this PR incompatible with master.

I'm not sure about @benedikt-voelkel general changes in this branch (see #708), maybe those can be merged already with master before getting out of sync with new developments. 

For what concerns reviving the dhadron case, this will come when the new MCs are ready and I start analysing Lc PbPb again. For the moment dhadronmult (with some small changes that Clara and Jaime already use) also works fine

**Pull request closed without merging**